src/custom.rs

```diff
@@ -90,10 +90,16 @@ macro_rules! register_custom_getrandom {
 }
 
 #[allow(dead_code)]
-pub fn getrandom_inner(dest: &mut [u8]) -> Result<(), Error> {
+pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
     extern "C" {
         fn __getrandom_custom(dest: *mut u8, len: usize) -> u32;
     }
+    // Previously we always passed a valid, initialized slice to
+    // `__getrandom_custom`. Ensure `dest` has been initialized for backward
+    // compatibility with implementations that rely on that (e.g. Rust
+    // implementations that construct a `&mut [u8]` slice from `dest` and
+    // `len`).
+    let dest = uninit_slice_fill_zero(dest);
```

Comment:
I think we might be able to remove this in a followup PR without a breaking change.

For example, if we were using the `custom` implementation:
  - `getrandom` just directly calls `__getrandom_custom`
  - `getrandom_uninit` calls `uninit_slice_fill_zero` then `__getrandom_custom`

But how exactly we do this should be discussed in a smaller, followup PR.

Reply:
It would remove zeroization for `getrandom`, but it still will be present in `getrandom_uninit`, which is not ideal. The idea is that in the next breaking release we will slightly change contract for `__getrandom_custom`.

I am not sure it's worth to introduce such divergences for the `custom` feature. In many cases compiler should be able eliminate zeroization for `getrandom` calls, since users often pass to it zero-initialized buffers.